common/src/lib.rs

```diff
@@ -156,22 +156,22 @@ impl DatabaseReadyInfo {
 /// Store that holds all the secrets available to a deployment
 #[derive(Deserialize, Serialize, Clone)]
 pub struct SecretStore {
-    pub(crate) secrets: BTreeMap<String, String>,
+    pub(crate) secrets: BTreeMap<String, Secret<String>>,
 }
 
 impl SecretStore {
-    pub fn new(secrets: BTreeMap<String, String>) -> Self {
+    pub fn new(secrets: BTreeMap<String, Secret<String>>) -> Self {
         Self { secrets }
     }
 
     pub fn get(&self, key: &str) -> Option<String> {
-        self.secrets.get(key).map(ToOwned::to_owned)
+        self.secrets.get(key).map(|k| ToOwned::to_owned(k.expose()))
     }
 }
 
 impl IntoIterator for SecretStore {
-    type Item = (String, String);
-    type IntoIter = <BTreeMap<String, String> as IntoIterator>::IntoIter;
+    type Item = (String, Secret<String>);
+    type IntoIter = <BTreeMap<String, Secret<String>> as IntoIterator>::IntoIter;
```

Comment:
Is this breaking for users? What do you think about providing the `into_inter` as before? (e.g. returning (String, String>)).

Reply:
It's breaking because of the change to the `SecretStore`. Indeed if this is what we expose to the user, we probably want to revert this part to how it was.